supervision/classification/core.py

```diff
@@ -0,0 +1,226 @@
+            annotations={name: self.annotations[name] for name in test_names},
+        )
+        return train_dataset, test_dataset
+
+    def as_multiclass_folder_structure(
+        self, root_directory_path: str, output_directory_path: str
+    ) -> None:
+        """
+        Saves the dataset as a multi-class folder structure.
+
+        Args:
+            root_directory_path (str): The path to the root directory of the images with which you are working.
+            output_directory_path (str): The path to the directory where the dataset will be saved.
+
+        Example:
+            ```python
+            >>> import supervision as sv
+
+            >>> cd = sv.ClassificationDataset(...)
+
+            >>> cd.as_multiclass_folder_structure(
+            ...     root_directory_path="./images/valid/",
+            ...     output_directory_path="./out",
+            ... )
+        """
+        if not os.path.exists(output_directory_path):
+            os.makedirs(output_directory_path)
+
+        for class_name in self.classes:
+            if not os.path.exists(os.path.join(output_directory_path, class_name)):
+                os.makedirs(os.path.join(output_directory_path, class_name))
+
+            for image in self.annotations:
+                if self.annotations[image].class_id[0] == self.classes.index(
+                    class_name
+                ):
+                    full_dir = os.path.join(root_directory_path, class_name, image)
+
+                    cv2.imwrite(
+                        os.path.join(output_directory_path, class_name, image),
+                        cv2.imread(full_dir),
+                    )
+
+    @classmethod
+    def from_multiclass_folder_structure(
+        cls, root_directory_path: str
+    ) -> "ClassificationDataset":
+        """
+        Load data from a multiclass folder structure into a ClassificationDataset.
+
+        Args:
+            root_directory_path (str): The path to the dataset directory.
+
+        Returns:
+            ClassificationDataset: The dataset.
+
+        Example:
+            ```python
+            >>> import supervision as sv
+
+            >>> cd = sv.ClassificationDataset.from_multiclass_folder_structure(
+            ...     root_directory_path="./dataset",
+            ... )
+        """
+        classes = os.listdir(root_directory_path)
+
+        classes = list(set(classes))
+        classes.sort()
+
+        images = {}
+        annotations = {}
+
+        for class_name in classes:
+            if not os.path.exists(os.path.join(root_directory_path, class_name)):
+                continue
+
+            for image in os.listdir(os.path.join(root_directory_path, class_name)):
+                images[image] = cv2.imread(
+                    os.path.join(root_directory_path, class_name, image)
+                )
+                annotations[image] = Classifications(
+                    class_id=np.array([classes.index(class_name)]),
```

Comment:
You can calculate `classes.index(class_name)` at the top of the first for loop, it will not get changed until next `class_name`.